### PR TITLE
Fix false positive `consider-using-dict-comprehension` when creating a dict using a list of tuple where key AND value vary depending on the same condition

### LINKED ISSUE #5588: False positive `consider-using-dict-comprehension` when creating a dict using a list of tuple where key AND value vary depending on the same condition
### Bug description

The following code snippet raises R1717:

```python
dict([(k, v) if v > 0 else (k, 0) for k, v in stuff.items()])
```

Both of these contain an `IfExp`, which I don't think can be converted into a dict comprehension. Please correct me if I'm wrong.

### Configuration

_No response_

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
************* Module mytest
mytest.py:1:0: R1717: Consider using a dictionary comprehension (consider-using-dict-comprehension)
```

### Expected behavior

R1717 to not show up

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6-dev0
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by tushar-deepsource**
## Type of Changes

Adds a condition where R1717 should not be raised

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #5588

### CODE REVIEW

**Comment by tushar-deepsource** [comment]
How am I still a first time contributor (:

**Comment by tushar-deepsource** [comment]
> * Tushar Sadhwani (tusharsadhwani): contributor

My personal account is in contributors. Should I add another entry?

**Comment by Pierre-Sassoulas** [comment]
> My personal account is in contributors. Should I add another entry?

No, but you can add an entry in https://github.com/PyCQA/pylint/blob/main/.copyrite_aliases

File: tests/functional/c/consider/consider_using_dict_comprehension.py [hunk 1]
```diff
@@ -8,5 +8,9 @@
 
 dict([(number, number*2) for number in numbers])  # [consider-using-dict-comprehension]
 
+stuff = {1: 10, 2: -20}
+dict([(k, v) if v > 0 else (k, 0) for k, v in stuff.items()])  # [consider-using-dict-comprehension]
```
Comment (Pierre-Sassoulas) [inline tests/functional/c/consider/consider_using_dict_comprehension.py:12]:
```suggestion
dict([(k, v) if v > 0 else (k, 0) for k, v in stuff.items()])  # [consider-using-dict-comprehension]
dict([(k, v) if v > 0 else (k*2, v) for k, v in stuff.items()])  # [consider-using-dict-comprehension]
```

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -934,9 +934,28 @@ def _check_consider_using_comprehension_constructor(self, node):
...
-            if node.func.name == "dict" and not isinstance(
-                node.args[0].elt, nodes.Call
-            ):
+            if node.func.name == "dict":
+                element = node.args[0].elt
+                if isinstance(element, nodes.Call):
+                    return
+
+                # If we have an `IfExp` here where both the key AND value
+                # are different, then don't raise the issue.
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/refactoring_checker.py:943]:
```suggestion
                # are different, then don't raise the issue. See #5588
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I'd add another test case for the case where the key change and not the value.

**Comment by tushar-deepsource** [comment]
@Pierre-Sassoulas need a workflow approval

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
Very nice ! Congratulation on becoming a contributor on your second account 😄 Would you mind adding an entry in the copyrite alias to be accounted as a single person (in this MR would work or another one, let me know) ?

**Comment by tushar-deepsource** [comment]
Done

**Comment by Pierre-Sassoulas** [comment]
Closing to relaunch github actions.

**Pull request merged**